src/main/java/org/junit/runners/Parameterized.java

```diff
@@ -276,10 +330,15 @@ private ParametersRunnerFactory getParametersRunnerFactory()
             }
         }
 
+        private Integer parameterCount;
+
         private TestWithParameters createTestWithNotNormalizedParameters(
                 String pattern, int index, Object parametersOrSingleParameter) {
             Object[] parameters = (parametersOrSingleParameter instanceof Object[]) ? (Object[]) parametersOrSingleParameter
                     : new Object[] { parametersOrSingleParameter };
+            if (parameterCount == null) {
+                parameterCount = parameters.length;
+            }
```

Comment:
If the number of parameters were different for some parameter sets, you would get a failure when the child runner is run.